src/JsonSchema/Constraints/ObjectConstraint.php

```diff
@@ -42,15 +42,23 @@ function check($element, $definition = null, $path = null, $additionalProp = nul
 
     public function validatePatternProperties($element, $path, $patternProperties)
     {
+        $try = array('/','#','+','~','%');
         $matches = array();
         foreach ($patternProperties as $pregex => $schema) {
+            // Choose delimiter. Necessary for patterns like ^/ , otherwise you get error
+            foreach ($try as $delimiter) {
```

Comment:
`$delimiter` should be defined and defaulted to a sensible value before this loop. Otherwise, `$delimiter` is only defined per the loop and will be set the last value if nothing breaks the loop.